### ISSUE TITLE
Cannot stringify arbitrary non-POJOs

### ISSUE THREAD

**Issue by Mflavien01** [wimluk/public-lib#1]
When I return the stream i have this error : 

> Error: Data returned from action inside /app is not serializable:

how can i fix this issue ?

**Comment by wimluk** [comment]
could you provide a little bit more context?

**Comment by Mflavien01** [comment]
I make like you in my code to receive stream and when i want to return data to the clientside i have this error 

> Error: Data returned from action inside /app is not serializable:

Is in this part of the code :

`const stream = await OpenAIStream(payload);`
  `return new Response(stream);`

**Comment by wimluk** [comment]
I just tried the code again and unfortunately, I could not reproduce the issue. Make sure to use the same versions of the dependencies and also provide a valid API key (make sure that you also have enough usage left, after your free credits expire you have to provide a payment method or it will not work). Also make sure that you provide the correct configurations.

**Comment by Mflavien01** [comment]
All is in the correct configurations, all work even this return, the issues is due to the type of the variable return, you return a "Uint8Array" variable is that ?

**Comment by wimluk** [comment]
With stream: true, in the payload, we define whether we want to get a stream back with partial progress. If set, tokens will be sent as data-only server-sent events (https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#Event_stream_format) as they become available.

The event stream consists of Server-Sent Events (SSE), which is a standard for sending real-time updates from the server to the client over a single HTTP connection. SSE uses UTF-8 encoding for text data to maintain compatibility with various clients and servers.

In our code, the response from the API is read as chunks using a for-await loop. Each chunk is then decoded from its binary form to a string using a TextDecoder instance, which handles the UTF-8 encoding. The decoded string is then passed to the parser, which processes the Server-Sent Events and invokes the onParse function for each parsed event.

Inside the ReadableStream, the onParse function is defined as a callback for the eventsource-parser. This function will be called when a new event is parsed from the API response. The function checks if the parsed event contains data and processes it accordingly.

The response from the API is read using an asynchronous for-await loop. Each chunk of the response body is decoded using a TextDecoder instance and then fed into the parser, which in turn calls the onParse function for each parsed event. When the onParse function is called, it checks if the event type is "event" and if it contains data. If so, it processes the data by parsing it as JSON, extracting the generated text, and then encoding the text into a typed array (Uint8Array) using a TextEncoder instance.

The generated text, now represented as a typed array, is enqueued into the ReadableStream's controller using the controller.enqueue() method. This makes the text available to the consumer of the stream. A ReadableStream supports both text and binary data as input. By encoding the text as a Uint8Array, you ensure that the ReadableStream can properly handle and process the data as a binary data stream and is more efficient.

In summary, the requirement of encoding the event stream using UTF-8 comes from the Server-Sent Events standard and ensures that the text data is represented consistently and can be processed by a wide range of systems and libraries.

You can check the following article for more information: https://web.dev/streams/

**Comment by Mflavien01** [comment]
OK, thanks! All of my code works, but as I mentioned earlier, I have a ReadableStream variable named stream like you in the file route.ts. The only issue is transitioning from the server side to the client side. I took the code from the ClientSection file to read the ReadableStream variable, and it works perfectly. I have all the words generated by OpenAI, so I don't understand what happens when I return the data at the end of the function.
